### PR TITLE
Database Changes - Add SentHearingAdminEmailEvent

### PR DESCRIPTION
**Pull request by odonnadieu**
Resolves [CASEFLOW-2212](https://vajira.max.gov/browse/CASEFLOW-2212)

### Description
These changes are based on these [tech spec](https://hackmd.io/fYAyoejJQGaSdj_Ph2xdKA?view#proposed-solution)

Added these columns to the sent_hearing_email_events table; all can be nil:

- `send_successful` (a boolean value that is nil if the status hasn't been checked or is "new" or "sending"; true if the status is "sent"; and false if the status is "failed", "inconclusive", "canceled", or "blacklisted")
- `send_successful_checked_at` (The date the status was last checked/updated)
- Created a new table `SentHearingAdminEmailEvent` to keep track of the 'administrative' email sent to our users.

### Acceptance Criteria
- [ ] Code compiles correctly

### Database Changes
*Only for Schema Changes*

* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [x] Run `make docs` (after running `make migrate`) to update DB schema docs
* [x] Post this PR in #appeals-schema with a summary

### CODE REVIEW

File: app/models/hearings/sent_hearing_email_event.rb [hunk 1]
```diff
@@ -11,6 +11,8 @@ class SentHearingEmailEvent < CaseflowRecord
   belongs_to :sent_by, class_name: "User"
   belongs_to :email_recipient, class_name: "HearingEmailRecipient"
 
+  has_many :sent_hearing_admin_email_events
```
Comment (tomas-nava) [inline app/models/hearings/sent_hearing_email_event.rb:14]:
```suggestion
  has_one :sent_hearing_admin_email_events
```
At the moment, we never expect to have more than one admin email per sent mail.

**Review by tomas-nava — APPROVED** [review]
It looks great! I have one blocking change, but when that's addressed, it can be merged.

File: app/models/hearings/sent_hearing_email_event.rb [hunk 2]
```diff
@@ -11,6 +11,8 @@ class SentHearingEmailEvent < CaseflowRecord
   belongs_to :sent_by, class_name: "User"
   belongs_to :email_recipient, class_name: "HearingEmailRecipient"
 
+  has_one :sent_hearing_admin_email_events
```
Comment (yoomlam) [inline app/models/hearings/sent_hearing_email_event.rb:14]: Shouldn't this be singular, as in `sent_hearing_admin_email_event`?

File: db/migrate/20210826144628_create_sent_hearing_admin_email_events.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateSentHearingAdminEmailEvents < Caseflow::Migration
+  def change
+    create_table :sent_hearing_admin_email_events do |t|
+      t.references :sent_hearing_email_event, foreign_key: true, index: false, comment: "Associated sent hearing email event."
```
Comment (yoomlam) [inline db/migrate/20210826144628_create_sent_hearing_admin_email_events.rb:4]: Why is `index: false`?

**Review by yoomlam — COMMENTED** [review]
Sorry for the late review. I was OOO.

**Pull request merged**